src/pages/today/index.tsx

```diff
@@ -25,32 +58,63 @@ export type HabitTodo = {
 export default function Today() {
 	const [date, setDate] = useState(new Date());
 	const [habitTodos, setHabitTodos] = useState<HabitTodo[]>([]);
+	const documentId = useRef<string>(null);
 
 	const completedHabits = habitTodos.filter((habit) => habit.isComplete);
 	const incompleteHabits = habitTodos.filter((habit) => !habit.isComplete);
 
 	const isDateToday = isToday(date);
 	const isDateYesterday = isYesterday(date);
 
+	const handleCheckHabit =
+		(habitId: string) => (e: React.ChangeEvent<HTMLInputElement>) =>
+			markHabitComplete(e.target.checked, habitId, documentId.current);
+
 	useEffect(() => {
-		const unsub = onSnapshot(
-			getDateDoc(date),
-			(doc) => {
-				if (doc.exists()) {
-					const { date, ...habits } = doc.data();
-					setHabitTodos(
-						Object.entries(habits).map(([id, habit]) => ({
-							id,
-							...(habit as Omit<HabitTodo, 'id'>),
-						}))
+		return onSnapshot(
+			query(
+				datesCollection(),
+				where('date', '>=', startOfDay(date)),
+				where('date', '<=', endOfDay(date)),
+				limit(1)
+			),
+			async ({ empty, docs }) => {
+				if (empty) {
+					const day = new Intl.DateTimeFormat('en-US', {
+						weekday: 'long',
+					}).format(date);
```

Comment:
it's a bit messy to put this logic in here in terms of readability, I'd extract it to a helper function and just do like `const day = currentDay()`

Reply:
and actually you're only using it on line 89, so you could just inline it there